### PR TITLE
Extended map to support geo location entities

### PR DESCRIPTION
**Pull request by exxamalte**
This is an extension to the lovelace map card that can not only display manually configured entities, but allows for configuring a `geo_location_sources` array; each geo location platform defines its source. All current geo location entities with that source will be displayed on the map automatically.
Either `entities` or `geo_location_sources` is required; if `geo_location_sources` is defined, then `entities` is optional.

Sample card configuration:
```
type: map
geo_location_sources:
  - nsw_rural_fire_service_feed
entities:
  - zone.home
default_zoom: 9
```

lovelace-geo-location-map

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io):** https://github.com/home-assistant/home-assistant.io/pull/7880

### CODE REVIEW

**Comment by exxamalte** [comment]
Hold on, I just realised that `entities` is required, but instead `source` should be required and `entities` optional.

**Comment by iantrich** [comment]
We have a map card. Why would you not at least add this source to the existing rather than copy all its code to a new file to add it?

**Comment by exxamalte** [comment]
Sure, happy to do that as well. I'm just wondering how to untangle which config parameters are then required and which ones are optional: The default map requires `entities` to be set, while the geo location map requires `source` to be set and additional `entities` are optional.

**Comment by exxamalte** [comment]
> Geo location sources should probably be an array?

Good point, because that would be the only way to join entities from different geo location platforms into one map.

File: src/panels/lovelace/cards/hui-map-card.js [hunk 1]
```diff
@@ -205,7 +222,24 @@ class HuiMapCard extends PolymerElement {
     }
     const mapItems = (this._mapItems = []);
 
-    this._configEntities.forEach((entity) => {
+    var allEntities = [];
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-map-card.js]: Doing use var

File: src/panels/lovelace/cards/hui-map-card.js [hunk 2]
```diff
@@ -111,7 +111,24 @@ class HuiMapCard extends PolymerElement {
       throw new Error("Error in card configuration.");
     }
 
-    this._configEntities = processConfigEntities(config.entities);
+    this._configGeoLocationSources = config.geo_location_sources;
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-map-card.js]: Don't assign before raising errors

**Review by balloob — CHANGES_REQUESTED** [review]

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
Do you want to submit a PR to add this functionality to a map demo ?

**Comment by exxamalte** [comment]
> Do you want to submit a PR to add this functionality to a map demo ?

Sure. I'll take a look at that now...

**Comment by frenck** [comment]
Just a note @exxamalte, `geolocation` is one word. So, it should be `geolocation_sources`...

**Comment by thomasloven** [comment]
@exxamalte I cannot find any documentation for this. Do you have something prepared, or should I add it when I'm documenting #2489 ?

**Comment by exxamalte** [comment]
@thomasloven The documentation for this PR can be found here: https://github.com/home-assistant/home-assistant.io/pull/7880

**Pull request merged**